s3storageprovider/src/main/java/org/duracloud/s3storage/S3StorageProvider.java

```diff
@@ -587,26 +587,11 @@ public String addContent(String spaceId,
 
         // Compare checksum
         String providerChecksum = getETagValue(etag);
-
-        try {
-            String checksum = wrappedContent.getMD5();
-            StorageProviderUtil.compareChecksum(providerChecksum,
-                                                spaceId,
-                                                contentId,
-                                                checksum);
-        } catch(ChecksumMismatchException e) {
-            try { // Clean up
-                s3Client.deleteObject(bucketName, contentId);
-            } catch (AmazonClientException amazonException) {
-                log.debug("Content item {} in space {} failed checksum test " +
-                          "and call to delete content item failed as well. " +
-                          "The content item was most likely never added to " +
-                          "storage.", contentId, spaceId);
-            }
-            // Rethrow to let the client know the file was not stored properly
-            throw e;
-        }
-
+        String checksum = wrappedContent.getMD5();
+        StorageProviderUtil.compareChecksum(providerChecksum,
```

Comment:
If the compareChecksum() call throws ChecksumMismatchException, there will not be a mention in the logs that this happened. Perhaps a log statement (WARN level) should be added either in a try/catch here, or in the method.